server/graphql/utils/request.ts

```diff
@@ -20,19 +22,61 @@ const getRequestsById = (requestIds, dataSources) => {
   return requestIds.map(id => dataSources.requests.getById(id))
 }
 
-const updateRequestHelper = (request, dataSources): Promise<ServerResponseInterface> => {
+const updateRequestHelper = async (request, dataSources): Promise<Document> => {
   if(!request.id) {
     throw new UserInputError('Missing argument value', { argumentName: 'id' })
   }
-  const requestTypeId = dataSources.requests.getById(request.id.toString()).requestType
-  return dataSources.requests.update(request)
-    .then(res => {
-      updateRequestTypeHelper({"id": requestTypeId}, dataSources)
-      return res
-    })
-    .catch(error => {
-      return error
-    })
+  const session = await mongoose.startSession()
+  try {
+    session.startTransaction()
+    const currentRequest = dataSources.requests.getById(request.id.toString())
+    const requestTypeId = currentRequest.requestType.toString()
+    const res = await dataSources.requests.update(request, session)
+    if(request.fulfilled === true && currentRequest.fulfilled === false) {
+      request.dateFulfilled = Date.now()
```

Comment:
Should we set `request.dateFulfilled = Date.now()` before we call `dataSources.requests.update(request, session)`?

Reply:
yup. i guess thats what happens when i copy paste code :facepalm: